r-package/grf/R/merge_forests.R

```diff
@@ -0,0 +1,78 @@
+#' Merges a list of forests that were grown using the same data into one large forest.
+#'
+#' @param forest_list A `list` of forests to be concatenated. 
+#'                        All forests must be of the same type, and the type must be a subclass of `grf`.
+#'                        
+#' @param compute.oob.predictions Whether OOB predictions on training set should be precomputed.
+#'        Note that even if OOB predictions have already been precomputed for the forests in 'forest_list',
+#'        those predictions are not used. Instead, a new set of oob predictions is computed anew using the 
+#'        larger forest.
+#' 
+#' @return A single forest containing all the trees in each forest in the input list.
+#'
+#' @examples \dontrun{
+#' # Train standard regression forests
+#' n = 50; p = 10
+#' X = matrix(rnorm(n*p), n, p)
+#' Y = X[,1] * rnorm(n)
+#' r.forest1 = regression_forest(X, Y, compute.oob.predictions = FALSE, num.trees = 100)
+#' r.forest2 = regression_forest(X, Y, compute.oob.predictions = FALSE, num.trees = 100)
+#'
+#' # Join the forests together. The resulting forest will contain 200 trees.
+#' big_rf = merge_forests(list(r.forest1, r.forest2))
+#' }
+#' 
+#' @export
+merge_forests <- function(forest_list, compute.oob.predictions=TRUE) {
+  
+  validate_forest_list(forest_list)
+  
+  first_forest <- forest_list[[1]]
+  data <- create_data_matrices(first_forest$X.orig, first_forest$Y.orig)
+  
+  big_forest <- merge(forest_list, data$default, data$sparse)
+  
+  big_forest[["ci.group.size"]] <- first_forest$ci.group.size
+  big_forest[["X.orig"]] <- first_forest$X.orig
+  big_forest[["Y.orig"]] <- first_forest$Y.orig
+  big_forest[["clusters"]] <- first_forest$clusters
+  big_forest[["min.node.size"]] <- first_forest$min.node.size
```

Comment:
Is `min.node.size` actually returned as part of the forest object?

Reply:
Interestingly, it's returned as part of `regression.forest`, but not `causal.forest`.

Reply:
I think we can get rid of it, it's inconsistent that we're adding this to the returned object.